src/main/kotlin/no/nav/syfo/aktivitetskrav/cronjob/JournalforAktivitetskravVarselCronjob.kt

```diff
@@ -0,0 +1,107 @@
+import no.nav.syfo.aktivitetskrav.AktivitetskravVarselService
+import no.nav.syfo.application.cronjob.Cronjob
+import no.nav.syfo.application.cronjob.CronjobResult
+import no.nav.syfo.client.dokarkiv.DokarkivClient
+import no.nav.syfo.client.dokarkiv.domain.*
+import no.nav.syfo.client.pdl.PdlClient
+import no.nav.syfo.domain.PersonIdent
+import org.slf4j.LoggerFactory
+
+class JournalforAktivitetskravVarselCronjob(
+    private val aktivitetskravVarselService: AktivitetskravVarselService,
+    private val dokarkivClient: DokarkivClient,
+    private val pdlClient: PdlClient,
+) : Cronjob {
+    override val initialDelayMinutes: Long = 2
+    override val intervalDelayMinutes: Long = 10
+
+    override suspend fun run() {
+        val result = runJob()
+        log.info(
+            "Completed journalføring of aktivitetskrav-varsel with result: {}, {}",
+            StructuredArguments.keyValue("failed", result.failed),
+            StructuredArguments.keyValue("updated", result.updated),
+        )
+    }
+
+    suspend fun runJob(): CronjobResult {
+        val result = CronjobResult()
+
+        val ikkeJournalforteAktivitetskravVarsel = aktivitetskravVarselService.getIkkeJournalforte()
+
+        ikkeJournalforteAktivitetskravVarsel.forEach { (personIdent, aktivitetskravVarsel, pdf) ->
+            try {
+                val navn = pdlClient.navn(personIdent = personIdent)
+                val journalpostRequest = createJournalpostRequest(
+                    personIdent = personIdent,
+                    navn = navn,
+                    pdf = pdf
+                )
+
+                dokarkivClient.journalfor(
+                    journalpostRequest = journalpostRequest,
+                ).also {
+                    aktivitetskravVarselService.updateJournalpostId(
+                        varsel = aktivitetskravVarsel,
+                        journalpostId = it.journalpostId.toString(),
+                    )
+                    result.updated++
+                }
+            } catch (e: Exception) {
+                log.error("Exception caught while attempting journalforing of aktivitetskrav-varsel", e)
+                result.failed++
+            }
+        }
+
+        return result
+    }
+
+    companion object {
+        private val log = LoggerFactory.getLogger(JournalforAktivitetskravVarselCronjob::class.java)
+    }
+}
+
+fun createJournalpostRequest(personIdent: PersonIdent, navn: String, pdf: ByteArray): JournalpostRequest {
+    val avsenderMottaker = AvsenderMottaker.create(
+        id = personIdent.value,
+        idType = BrukerIdType.PERSON_IDENT,
+        navn = navn,
+    )
+    val bruker = Bruker.create(
+        id = personIdent.value,
+        idType = BrukerIdType.PERSON_IDENT,
+    )
+
+    // På sikt avhengig av om bruker er reservert mot digital kommunikasjon eller ikke (må lagres med AktivitetskravVarsel?)
+    val kanal = JournalpostKanal.SENTRAL_UTSKRIFT
```

Comment:
Det er vel best å sjekke KRR on demand når vi skal begynneå bry oss om det? (Med cache da sikkert) 🤔

Reply:
Ja, vi må vel sjekke KRR på et tidspunkt når vi skal slutte med kun fysisk brev. 
Har sett i `isdialogmote` at vi sjekker det når varselet lages og så persisterer det med varselet (digitalt `true/false`).
En fordel med det er jo at man slipper å sjekke på nytt når det journalføres (og evt i egen jobb for å produsere til esyfovarsel). Samt at det kanskje er kjekt å vite om et varsel ble sendt digitalt eller ei (om bruker er digital eller ikke kan jo endre seg), Tenker at det er tilstanden når veileder oppretter varselet som bør gjelde :thinking: